hoomd/md/Thermostat.h

```diff
@@ -337,38 +337,131 @@ class MTTKThermostat : public Thermostat
 class BussiThermostat : public Thermostat
     {
     public:
+    /** Construct the thermostat.
+
+        @param T Temperature set point over time.
+        @param group Group of particles this thermostat is applied to.
+        @param thermo Use to compute the thermodynamic properties of the group.
+        @param sysdef Used to access the simulation seed and MPI communicator.
+        @param tau Thermostat time constant.
+    */
     BussiThermostat(std::shared_ptr<Variant> T,
                     std::shared_ptr<ParticleGroup> group,
                     std::shared_ptr<ComputeThermo> thermo,
-                    std::shared_ptr<SystemDefinition> sysdef)
-        : Thermostat(T, group, thermo, sysdef)
+                    std::shared_ptr<SystemDefinition> sysdef,
+                    Scalar tau)
+        : Thermostat(T, group, thermo, sysdef), m_tau(tau)
         {
         }
 
     std::array<Scalar, 2> getRescalingFactorsOne(uint64_t timestep, Scalar deltaT) override
         {
         m_thermo->compute(timestep);
 
-        const auto ntdof = m_thermo->getTranslationalDOF();
-        const auto nrdof = m_thermo->getRotationalDOF();
+        Scalar time_decay_factor;
+        Scalar t_random_left;
+        Scalar r_random_left;
+        // Integration methods will always operate on subsets with integer degrees of freedom.
+        // Partial degrees of freedom are removed by getTranslationalDOF() when the subset of
+        // particles is not the same subet selected for integration.
+        const int ntdof = static_cast<int>(m_thermo->getTranslationalDOF());
+        const int nrdof = static_cast<int>(m_thermo->getRotationalDOF());
+        const auto ket_int = m_thermo->getTranslationalKineticEnergy();
+        const auto ker_int = m_thermo->getRotationalKineticEnergy();
         if ((ntdof != 0 && m_thermo->getTranslationalKineticEnergy() == 0)
             || (nrdof != 0 && m_thermo->getRotationalKineticEnergy() == 0))
             {
             throw std::runtime_error("Bussi thermostat requires non-zero initial temperatures");
             }
+
         unsigned int instance_id = 0;
         if (m_group->getNumMembersGlobal() > 0)
             instance_id = m_group->getMemberTag(0);
         RandomGenerator rng(Seed(RNGIdentifier::BussiThermostat, timestep, m_sysdef->getSeed()),
                             instance_id);
 
         const auto set_T = m_T->operator()(timestep);
-        GammaDistribution<double> gamma_translation(ntdof / 2.0, set_T);
-        GammaDistribution<double> gamma_rotation(nrdof / 2.0, set_T);
 
-        return {std::sqrt(gamma_translation(rng) / m_thermo->getTranslationalKineticEnergy()),
-                std::sqrt(gamma_rotation(rng) / m_thermo->getRotationalKineticEnergy())};
+        if (m_tau != 0.0)
+            {
+            time_decay_factor = exp(-deltaT / m_tau);
+            }
+        else
+            {
+            time_decay_factor = 0.0;
+            }
```

Comment:
@joaander It seems to me that it will be problematic when both deltaT and m_tau are zero when calculating exp(-deltaT / m_tau)

Reply:
You could add:
```
if (deltaT == 0.0)
    {
    return {1.0, 1.0};
    }
```
as a clear special case at the top of the function.